Crash report 2.9.1: NPE in StatsService.onStartCommand

```
java.lang.RuntimeException: Unable to start service org.wordpress.android.ui.stats.service.StatsService@4118a560 with null: java.lang.NullPointerException
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2867)
       at android.app.ActivityThread.access$1900(ActivityThread.java:165)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1470)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.app.ActivityThread.main(ActivityThread.java:5370)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at org.wordpress.android.ui.stats.service.StatsService.onStartCommand(StatsService.java:101)
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2850)
       at android.app.ActivityThread.access$1900(ActivityThread.java:165)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1470)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.app.ActivityThread.main(ActivityThread.java:5370)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:525)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

--- a/src/org/wordpress/android/ui/stats/service/StatsService.java
+++ b/src/org/wordpress/android/ui/stats/service/StatsService.java
@@ -98,6 +98,17 @@ public class StatsService extends Service {
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
+        if (intent == null) {
+            AppLog.d(T.STATS, "StatsService was killed and restarted with a null intent.");
+            // if this service's process is killed while it is started (after returning from onStartCommand(Intent, int, int)),
+            // then leave it in the started state but don't retain this delivered intent.
+            // Later the system will try to re-create the service.
+            // Because it is in the started state, it will guarantee to call onStartCommand(Intent, int, int) after creating the new service instance;
+            // if there are not any pending start commands to be delivered to the service, it will be called with a null intent object.
+            stopRefresh();
+            return START_NOT_STICKY;
+        }
+
         final String blogId = StringUtils.notNullStr(intent.getStringExtra(ARG_BLOG_ID));
         final String currentServiceBlogId = getServiceBlogId();
 

